### PR TITLE
Laboratory work #4, Isakov Danila - 21FPL2

### PR DESCRIPTION
**Pull request by vchemsmisl**

### CODE REVIEW

File: lab_3_keywords_textrank/assets/report.csv [hunk 1]
```diff
@@ -0,0 +1,5 @@
+name,culture,business,crime,fashion,health,politics,science,sports,tech
```
Comment (artyomtugaryov) [inline lab_3_keywords_textrank/assets/report.csv:1]: remove

File: lab_3_keywords_textrank/main.py [hunk 2]
```diff
@@ -4,6 +4,11 @@
 """
 from pathlib import Path
 from typing import Optional, Union
+import csv
```
Comment (artyomtugaryov) [inline lab_3_keywords_textrank/main.py:7]: ? Revert

File: lab_4_summarization_textrank/main.py [hunk 3]
```diff
@@ -2,14 +2,21 @@
...
     TextPreprocessor
 
 PreprocessedSentence = tuple[str, ...]
 EncodedSentence = tuple[int, ...]
 
+def check_iter(iterable: Iterable, val_type: type) -> bool:
+    """
+    Checks, if all values in iterable are of needed type
+    """
+    if not all(isinstance(val, val_type) for val in iterable):
```
Comment (artyomtugaryov) [inline lab_4_summarization_textrank/main.py:17]: you can remove the result of `not all(isinstance(val, val_type) for val in iterable)` as is

File: lab_4_summarization_textrank/start.py [hunk 4]
```diff
@@ -26,6 +28,10 @@
         idf = json.load(file)
 
     paths_to_texts = [str(path) for path in TEXTS_PATH.glob('*.txt')]
+    preprocess = SentencePreprocessor(stop_words, tuple(sym for sym in punctuation))
```
Comment (artyomtugaryov) [inline lab_4_summarization_textrank/start.py:31]: only 4?

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

**Comment by artyomtugaryov** [comment]
Please, fix the previous comments

File: lab_4_summarization_textrank/main.py (hunk 3 above)
Comment (vchemsmisl) [reply lab_4_summarization_textrank/main.py]: Done

File: lab_4_summarization_textrank/main.py [hunk 5]
```diff
@@ -229,7 +344,14 @@ def update_vertex_score(
         :param scores: current vertices scores
         :return:
         """
-        pass
+        if not (vertex and isinstance(vertex, Sentence) and incidental_vertices and
```
Comment (demid5111) [inline lab_4_summarization_textrank/main.py:347]: refactor?

File: lab_4_summarization_textrank/main.py [hunk 6]
```diff
@@ -193,16 +295,23 @@ def get_similarity_score(self, sentence: Sentence, other_sentence: Sentence) ->
...
         """
         Updates graph instance with vertices and edges extracted from sentences
         :param sentences
         :return:
         """
-        pass
-
+        if not (sentences and isinstance(sentences, tuple) and check_iter(sentences, Sentence)):
+            raise ValueError
+        pairs = list(permutations(sentences, r=2))
```
Comment (demid5111) [inline lab_4_summarization_textrank/main.py:312]: replace with own

File: lab_4_summarization_textrank/main.py [hunk 7]
```diff
@@ -257,16 +379,24 @@ def get_top_sentences(self, n_sentences: int) -> tuple[Sentence, ...]:
...
         :return: summary
         """
-        pass
 
+        if not (n_sentences and isinstance(n_sentences, int) and not isinstance(n_sentences, bool)):
+            raise ValueError
+        sent_positions = {sent.get_text(): sent.get_position() for sent in self.get_top_sentences(n_sentences)}
+        sent_sorted = sorted(sent_positions, key=lambda x: sent_positions[x])
+        summary = '\n'.join(sent_sorted)
+        summary.replace('\n \n', '')
```
Comment (demid5111) [inline lab_4_summarization_textrank/main.py:398]: ?

**Review by demid5111 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**